django/middleware/csrf.py

```diff
@@ -154,15 +157,35 @@ def process_view(self, request, callback, callback_args, callback_kwargs):
                 if referer is None:
                     return self._reject(request, REASON_NO_REFERER)
 
+                referer = urlparse(referer)
+
+                # Make sure we have a valid URL for Referer.
+                if '' in (referer.scheme, referer.netloc):
+                    return self._reject(request, REASON_MALFORMED_REFERER)
+
+                # Ensure that our Referer is also secure.
+                if referer.scheme != 'https':
+                    return self._reject(request, REASON_INSECURE_REFERER)
```

Comment:
Yes, we should definitely reject HTTPS requests with no referrer; otherwise we may as well just remove the referrer-checking entirely. I don't think this PR changes that.